packages/psss-server/src/models/PsssSession.ts

```diff
@@ -4,19 +4,107 @@
  * Copyright (c) 2017 - 2020 Beyond Essential Systems Pty Ltd
  */
 
+import jwt from 'jsonwebtoken';
 import { DatabaseModel, DatabaseType } from '@tupaia/database';
 import { AccessPolicy } from '@tupaia/access-policy';
 
+import { AccessPolicyObject, SessionColumns, SessionDetails } from '../types';
+import { AuthConnection } from '../connections/AuthConnection';
+
+const getTokenExpiry = (accessToken: string) => {
+  const { exp: expiryAuthServerClock, iat: issuedAtAuthServerClock } = jwt.decode(accessToken);
+  // subtract 3 seconds to account for latency since generation on the auth server
+  const validForSeconds = expiryAuthServerClock - issuedAtAuthServerClock - 3;
+  const expiryPsssServerClock = Date.now() + validForSeconds * 1000;
+  return expiryPsssServerClock;
+};
+
 export class PsssSessionType extends DatabaseType {
+  id: string;
+
+  email: string;
+
+  access_policy: AccessPolicyObject;
+
+  access_token: string;
+
+  access_token_expiry: number;
+
+  refresh_token: string;
+
+  authConnection: AuthConnection;
+
+  refreshAccessTokenPromise: Promise<void> | null;
+
   static databaseType = 'psss_session';
 
+  constructor(model: PsssSessionModel, fieldValues: SessionColumns) {
+    super(model, fieldValues);
+
+    // explicitly reassign all field values to satisfy typescript
```

Comment:
It would be good to go around this at some point; but it's fine for now

Reply:
Totally. I wish we didn't need to, but I also understand that TypeScript can't just come up with types out of thin air.